pyro/contrib/funsor/handlers/plate_messenger.py

```diff
@@ -154,3 +156,174 @@ def _pyro_param(self, msg):
         frame = CondIndepStackFrame(self.name, None, self.size, self._counter)
         msg["cond_indep_stack"] = (frame,) + msg["cond_indep_stack"]
         msg["scale"] = msg["scale"] * self._scale
+
+
+class VectorizedMarkovMessenger(NamedMessenger):
+    """
+    Construct for Markov chain of variables designed for efficient elimination of Markov
+    dimensions using the parallel-scan algorithm. Whenever permissible, `pyro.vectorized_markov`
+    is interchangeable with `pyro.markov`.
+
+    The for loop generates both `int` and 1-dimensional :class:`torch.Tensor` indices:
+    (`0, ..., history-1, torch.arange(0, size-history), ..., torch.arange(history, size)`).
+    `int` indices are used to initiate the Markov chain and :class:`torch.Tensor` indices
+    are used to construct vectorized transition probabilities for efficient elimination by
+    the parallel-scan algorithm.
+
+    When `history==0` `pyro.vectorized_markov` behaves like a `pyro.plate`.
+    After the for loop is run, Markov variables are identified and then the `step`
+    information is constructed and added to the trace. `step` informs inference algorithms
+    which variables belong to a Markov chain.
+
+    .. code-block:: py
```

Comment:
Have you run `make docs` locally to check that your docstrings render correctly?